.github/workflows/typescript.yml

```diff
@@ -0,0 +1,24 @@
+name: Typescript
+
+on:
+  workflow_call:
+
+jobs:
+  tsc:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: "22"
+          cache: "npm"
+
+      - name: Install JS dependencies
+        run: npm ci
+
+      - name: Build the project
+        run: npm run build
```

Comment:
Genuine question, why run build instead of just running type checking (given this is titled `TypeScript`)?

Reply:
Just to note down what we discussed, this is the command that users of this library will most likely be running to do that actual building and it also hides any complexity that the build command might have.

Reply:
Yea, my read is that it runs type checking _and_ does the transpilation step the the tsc build does, so it's a more thorough test than just type checking alone. My only nitpick then would be to rename the workflow to "Run build" or "Build and type check" for a bit more clarity.

I won't hold up merging for that change though